File: client/FireSale/src/app/auctions/models/createAuctionDto.ts
import { TagDto } from 'src/app/shared/models/tagDto';
import { CreateImageDTO } from './createImageDto';

export interface CreateAuctionDTO {
  id: number;
  name: string;
  description: string;
  minimalBid: number;
  startDate: Date;
  endDate: Date;
  categories: number[];
  images: CreateImageDTO[];
  tags: TagDto[];
}

File: api/src/main/java/com/FireSale/api/dto/auction/CreateAuctionDTO.java
package com.FireSale.api.dto.auction;

import com.FireSale.api.dto.BaseDTO;
import com.FireSale.api.dto.TagDTO;
import lombok.Data;

import java.time.LocalDateTime;
import java.util.Collection;

@Data
public class CreateAuctionDTO extends BaseDTO {
    private String name;
    private String description;
    private LocalDateTime startDate;
    private LocalDateTime endDate;
    private Double minimalBid;
    private Collection<CreateImageDTO> images;
    private Collection<Long> categories;
    private Collection<TagDTO> tags;
}
